includes/class-sensei-data-cleaner.php

```diff
@@ -61,4 +76,30 @@ private static function cleanup_custom_post_types() {
 			}
 		}
 	}
+
+	/**
+	 * Cleanup data for taxonomies.
+	 *
+	 * @access private
+	 */
+	private static function cleanup_taxonomies() {
+		global $wpdb;
+
+		foreach ( self::$taxonomies as $taxonomy ) {
+			$terms = $wpdb->get_results(
+				$wpdb->prepare(
+					"SELECT term_id, term_taxonomy_id FROM $wpdb->term_taxonomy WHERE taxonomy = %s",
+					$taxonomy
+				)
+			);
+
+			// Delete all data for each term.
+			foreach ( $terms as $term ) {
+				$wpdb->delete( $wpdb->term_relationships, array( 'term_taxonomy_id' => $term->term_taxonomy_id ) );
+				$wpdb->delete( $wpdb->term_taxonomy, array( 'term_taxonomy_id' => $term->term_taxonomy_id ) );
+				$wpdb->delete( $wpdb->terms, array( 'term_id' => $term->term_id ) );
+				$wpdb->delete( $wpdb->termmeta, array( 'term_id' => $term->term_id ) );
+			}
+		}
+	}
```

Comment:
Is there a reason you aren't using `get_terms()` and `wp_delete_term()` here? Are they available in `uninstall.php`?

Reply:
I was inspired by https://github.com/easydigitaldownloads/easy-digital-downloads/blob/master/uninstall.php#L48 which uses DB queries, and also a little bit concerned about performance (the code for `wp_delete_term` is [over 70 lines long](https://developer.wordpress.org/reference/functions/wp_delete_term/) which seems pretty heavy compared to a few DB queries).

But other than that, no really compelling reasons. Thinking about it again, I'm leaning toward using the functions. That will match what I'm doing for the rest of the data deletion PR's, and it's a little less scary than raw DB queries. I'll ping you for another look when I've made that change 🙂

Reply:
It is a bit of a beast, but a lot of that is documentation. One nice thing about `wp_delete_term()` is it takes care of cache clearing, too.